Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -21,48 +21,72 @@ namespace ROS2::Utils
     {
         AZStd::vector<AZStd::string> modifiedLinks;
         using namespace AZ::rapidxml;
-        AZStd::vector<xml_node<>*> links;
+
         for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
         {
-            if (!link->first_node("inertial"))
+            bool modified = false;
+            bool inertiaPresent = true;
+
+            auto name_xml = link->first_attribute("name");
+            AZStd::string name = "unknown_link";
+            if (name_xml)
             {
-                links.push_back(link);
+                name = name_xml->value();
             }
-        }
 
-        for (auto* link : links)
-        {
-            xml_node<>* inertial = urdf->document()->allocate_node(node_element, "inertial");
+            auto inertial = link->first_node("inertial");
+            if (!inertial)
+            {
+                AZ_Warning("URDF", false, "Missing inertial tag in link %s, applying default tag.", name.c_str());
+                inertial = urdf->document()->allocate_node(node_element, "inertial");
+                link->append_node(inertial);
+                modified = true;
+                inertiaPresent = false;
+            }
 
-            xml_node<>* mass = urdf->document()->allocate_node(node_element, "mass");
-            mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
-            inertial->append_node(mass);
-            xml_node<>* inertia = urdf->document()->allocate_node(node_element, "inertia");
+            auto mass = inertial->first_node("mass");
+            if (!mass)
+            {
+                if(inertiaPresent)
+                {
+                    AZ_Warning("URDF", false, "Missing mass tag inside the inertial tag link %s, applying default mass tag.", name.c_str());
+                }
+                mass = urdf->document()->allocate_node(node_element, "mass");
+                mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
+                inertial->append_node(mass);
+                modified = true;
+            }
 
-            inertia->append_attribute(urdf->document()->allocate_attribute("ixx", "1."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("ixy", "0."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("ixz", "0."));
+            auto inertia = inertial->first_node("inertia");
+            if (!inertia)
+            {
+                if(inertiaPresent)
+                {
+                    AZ_Warning("URDF", false, "Missing inertia tag inside the inertial tag link %s, applying default inertia tag.", name.c_str());
+                }
```

Comment:
You could simplify a bit here too:
```suggestion
                AZ_Warning("URDF", !inertiaPresent, "Missing inertia tag inside the inertial tag link %s, applying default inertia tag.", name.c_str());
```